alicloud/resource_alicloud_network_acl_attachment.go

```diff
@@ -0,0 +1,156 @@
+	return &schema.Resource{
+		Create: resourceAliyunNetworkAclAttachmentCreate,
+		Read:   resourceAliyunNetworkAclAttachmentRead,
+		Delete: resourceAliyunNetworkAclAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"network_acl_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"resources": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"resource_type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				MaxItems: 20,
+			},
+		},
+	}
+}
+
+func resourceAliyunNetworkAclAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	networkAclService := NetworkAclService{client}
+
+	request := vpc.CreateAssociateNetworkAclRequest()
+	request.RegionId = client.RegionId
+	request.NetworkAclId = d.Get("network_acl_id").(string)
+	resources := []vpc.AssociateNetworkAclResource{}
+	vpcResource := []vpc.Resource{}
+	for _, e := range d.Get("resources").([]interface{}) {
+		res := e.(map[string]interface{})
+		source := vpc.AssociateNetworkAclResource{
+			ResourceId:   res["resource_id"].(string),
+			ResourceType: res["resource_type"].(string),
+		}
+		resources = append(resources, source)
+		vpcSource := vpc.Resource{
+			ResourceId:   res["resource_id"].(string),
+			ResourceType: res["resource_type"].(string),
+		}
+		vpcResource = append(vpcResource, vpcSource)
+	}
+	request.Resource = &resources
+	raw, err := client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
+		return vpcClient.AssociateNetworkAcl(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_network_acl_attachment", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName, raw)
+	d.SetId(request.NetworkAclId)
+
+	if err := networkAclService.WaitForNetworkAclAttachment(request.NetworkAclId, vpcResource, Available, DefaultTimeout); err != nil {
+		return WrapError(err)
+	}
+	return resourceAliyunNetworkAclAttachmentRead(d, meta)
+}
+
+func resourceAliyunNetworkAclAttachmentRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	networkAclService := NetworkAclService{client}
+
+	object, err := networkAclService.DescribeNetworkAcl(d.Id())
+	vpcResource := []vpc.Resource{}
+	for _, e := range d.Get("resources").([]interface{}) {
+		res := e.(map[string]interface{})
+		vpcSource := vpc.Resource{
+			ResourceId:   res["resource_id"].(string),
+			ResourceType: res["resource_type"].(string),
+		}
+		vpcResource = append(vpcResource, vpcSource)
+	}
+	err = networkAclService.DescribeNetworkAclAttachment(object.NetworkAclId, vpcResource)
```

Comment:
I think the line 93 should be moved into the method `DescribeNetworkAclAttachment ` and this method should return a ACL Object.